### PR TITLE
Bug 1977595: pseudo translation missing on OperatorHub page

### PR DESCRIPTION
**Pull request by florkbr**
Resolved remaining issues with the operatorhub page missing translations. From the bug:

> some strings not coming from API are not pesudo translated, for example
> - text 'All Items', 'N times' 'Filter by keyword'
> - text 'Install state', 'Installed', 'Not Installed'

https://bugzilla.redhat.com/show_bug.cgi?id=1977595

Screen Shot 2021-07-08 at 3 04 57 PM
Screen Shot 2021-07-08 at 3 05 18 PM

### CODE REVIEW

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx [hunk 1]
```diff
@@ -93,17 +94,14 @@ const operatorHubFilterGroups = [
...
-// t('olm~Install state')
-// t('olm~Capability level')
-// t('olm~Infrastructure features')
-const operatorHubFilterMap = {
-  catalogSourceDisplayName: 'Source',
-  provider: 'Provider',
-  installState: 'Install state',
-  capabilityLevel: 'Capability level',
-  infraFeatures: 'Infrastructure features',
+const getOperatorHubFilterMap = () => {
```
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx:97]: If this is only used in one component, it's probably simpler just to move the object into that component.

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx [hunk 2]
```diff
@@ -293,6 +277,25 @@ const determineAvailableFilters = (initialFilters, items, filterGroups) => {
     );
   });
 
+  // Always show both install state filters
+  if (!filters.installState) {
```
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx:281]: Can you explain the need to change the logic here? Previously, we always set the `installState` filters without this check. I'm assuming it's overwriting the value for `active`, but I'm not sure what's changed in the PR to cause that.

**Review by spadgett — COMMENTED** [review]
Thanks @florkbr

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx (hunk 2 above)
Comment (florkbr) [reply frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx:281]: Previously we set `filters.installState` at the top of this function, then iterated over each filter group passed in to initialize the filters with the possible values. I attempted to change the `Installed` and `Not Installed` labels in this object, but found that the change was not taking effect in the UI.

Turns out we pass `installState` in as a filter group (though I'm assuming this logic was added as that is not always the case). This means that the logic [here](https://github.com/openshift/console/pull/9450/files#diff-f5c2166645f6d9f295216d0a8e637a7e76d633490cbf3a4b94ed3aed11c5d27dR264) and [here](https://github.com/openshift/console/pull/9450/files#diff-f5c2166645f6d9f295216d0a8e637a7e76d633490cbf3a4b94ed3aed11c5d27dR276) was overwriting the initial `filters.installState` we set at the top of the function.

It was simpler/safer IMO to move the check below the loop. I can simplify this logic if `Installed` and `Not Installed` are the only two possible values for `installState`.

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx (hunk 1 above)
Comment (florkbr) [reply frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-items.tsx]: @spadgett done!

**Review by spadgett — APPROVED** [review]
/lgtm

**Pull request merged**